[SDL2] SYSTEM_LIBRARIES missing

Many users are having problems with SDL2 build because SYSTEM_LIBRARIES are not being sent.

On mingw-w64 static the error is:
```
-- Using msys root at C:/vcpkg-test/downloads/tools/msys2/c809757c94447846
-- Fixing pkgconfig file: C:/vcpkg-test/packages/sdl2_x64-mingw-static/lib/pkgconfig/sdl2.pc
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:235 (message):
  Unhandled string "-mwindows" was found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file or add the case to vcpkg_fixup_pkgconfig
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:299 (vcpkg_fixup_pkgconfig_check_files)
  ports/sdl2/portfile.cmake:101 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:136 (include)


Error: Building package sdl2:x64-mingw-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: sdl2:x64-mingw-static
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above
```

I can only build the SDL2 if I put the required libraries for mingw
`SYSTEM_LIBRARIES dbus-1 mwindows mingw32 dxerr8 m SDL2dmain SDL2
`

But, this is not the correct way to fix, other related issues show many systems with the same problem.
Is it possible make vcpkg_fixup_pkgconfig auto add these libraries required from all systems?

Related issues:
https://github.com/microsoft/vcpkg/issues/14598
https://github.com/microsoft/vcpkg/issues/13963
https://github.com/microsoft/vcpkg/issues/12998

> I mean `-mwindows` should be in IGNORE_FLAGS as it's not related to library dependencies.
> Your `vcpkg_fixup_pkgconfig` should be like this:
> 
> ```
> vcpkg_fixup_pkgconfig(
>     IGNORE_FLAGS "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/lib/pkgconfig/../../lib" "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/../../lib" "-Wl,--enable-new-dtags" "-Wl,--no-undefined" "-Wl,-undefined,error" "-Wl,-compatibility_version,${DYLIB_COMPATIBILITY_VERSION}" "-Wl,-current_version,${DYLIB_CURRENT_VERSION}" "-Wl,-weak_framework,Metal" "-Wl,-weak_framework,QuartzCore" "-mwindows"
>     SYSTEM_LIBRARIES dbus-1 mingw32 dxerr8 m
> )
> ```
> 
> If you can, please create a PR, I'm working on getting #14556 merged at the moment
> 
> For someone working on `vcpkg_fixup_pkgconfig`: Can we only check flags starting with `-l`? This would prevent having to specify a bunch of flags like what happened in `sdl2`.

This change not work, SDL2 libraries is required.
```
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:233 (message):
  Library "SDL2dmain" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
```

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -71,9 +71,7 @@ if(NOT VCPKG_CMAKE_SYSTEM_NAME)
 
     file(GLOB SHARE_FILES ${CURRENT_PACKAGES_DIR}/share/sdl2/*.cmake)
     foreach(SHARE_FILE ${SHARE_FILES})
-        file(READ "${SHARE_FILE}" _contents)
-        string(REPLACE "lib/SDL2main" "lib/manual-link/SDL2main" _contents "${_contents}")
-        file(WRITE "${SHARE_FILE}" "${_contents}")
+        vcpkg_replace_string("${SHARE_FILE}" "lib/SDL2main" "lib/manual-link/SDL2main")
     endforeach()
 endif()
 
@@ -94,6 +92,7 @@ if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
 endif()
 
 vcpkg_fixup_pkgconfig(
-    IGNORE_FLAGS "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/lib/pkgconfig/../../lib" "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/../../lib" "-Wl,--enable-new-dtags" "-Wl,--no-undefined" "-Wl,-undefined,error" "-Wl,-compatibility_version,${DYLIB_COMPATIBILITY_VERSION}" "-Wl,-current_version,${DYLIB_CURRENT_VERSION}" "-Wl,-weak_framework,Metal" "-Wl,-weak_framework,QuartzCore"
-    SYSTEM_LIBRARIES dbus-1
+    IGNORE_FLAGS
+        "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/lib/pkgconfig/../../lib" "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/../../lib" "-Wl,--enable-new-dtags" "-Wl,--no-undefined" "-Wl,-undefined,error" "-Wl,-compatibility_version,${DYLIB_COMPATIBILITY_VERSION}" "-Wl,-current_version,${DYLIB_CURRENT_VERSION}" "-Wl,-weak_framework,Metal" "-Wl,-weak_framework,QuartzCore" "-mwindows"
+    SYSTEM_LIBRARIES dbus-1 mingw32 dxerr8 m
 )

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -89,7 +89,8 @@ string(REGEX REPLACE ${DYLIB_COMPATIBILITY_VERSION_REGEX} "\\1" DYLIB_COMPATIBIL
 string(REGEX REPLACE ${DYLIB_CURRENT_VERSION_REGEX} "\\1" DYLIB_CURRENT_VERSION "${DYLIB_CURRENT_VERSION}")
 
 if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-	vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2" "-lSDL2d")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2main" "-lSDL2maind")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2 " "-lSDL2d ")
 endif()
 
 vcpkg_fixup_pkgconfig(

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -91,8 +91,4 @@ if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2 " "-lSDL2d ")
 endif()
 
-vcpkg_fixup_pkgconfig(
-    IGNORE_FLAGS
-        "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/lib/pkgconfig/../../lib" "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/../../lib" "-Wl,--enable-new-dtags" "-Wl,--no-undefined" "-Wl,-undefined,error" "-Wl,-compatibility_version,${DYLIB_COMPATIBILITY_VERSION}" "-Wl,-current_version,${DYLIB_CURRENT_VERSION}" "-Wl,-weak_framework,Metal" "-Wl,-weak_framework,QuartzCore" "-mwindows"
-    SYSTEM_LIBRARIES dbus-1 mingw32 dxerr8 m
-)
+vcpkg_fixup_pkgconfig()

--- a/ports/sdl2/vcpkg.json
+++ b/ports/sdl2/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "sdl2",
   "version-string": "2.0.14",
+  "port-version": 1,
   "description": "Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.",
   "homepage": "https://www.libsdl.org/download-2.0.php",
   "features": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5194,7 +5194,7 @@
     },
     "sdl2": {
       "baseline": "2.0.14",
-      "port-version": 0
+      "port-version": 1
     },
     "sdl2-gfx": {
       "baseline": "1.0.4",

--- a/versions/s-/sdl2.json
+++ b/versions/s-/sdl2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8d5fa523a69780e41605b585788d212d688f3c71",
+      "version-string": "2.0.14",
+      "port-version": 1
+    },
+    {
       "git-tree": "10ed256500fea4cd6cc5d3e6c6b5f2e9e364d78b",
       "version-string": "2.0.14",
       "port-version": 0

